talisker/wsgi.py

```diff
@@ -552,11 +571,40 @@ def __call__(self, environ, start_response):
             environ[config.wsgi_id_header] = str(uuid.uuid4())
         rid = environ[config.wsgi_id_header]
         environ['REQUEST_ID'] = rid
-        Context.request_id = rid
+        # needs to be different from request id, as request can be passed on to
+        # upstream services
+        environ['SENTRY_ID'] = uuid.uuid4().hex
 
-        # populate default values
+        Context.request_id = rid
         Context.current.soft_timeout = config.soft_request_timeout
 
+        # calculate ip route
+        route = None
+        try:
+            forwarded = environ.get('HTTP_X_FORWARDED_FOR')
+            if forwarded:
+                route = [a.strip() for a in forwarded.split(',')]
+            elif "REMOTE_ADDR" in environ:
+                route = [environ["REMOTE_ADDR"]]
+        except Exception as e:
+            logger.exception(e)
+        else:
+            if route is not None:
+                environ['ACCESS_ROUTE'] = route
+                environ['CLIENT_ADDR'] = route[-1]
```

Comment:
X-Forwarded-For is left to right, client, proxy1, proxy2 etc so this is getting the IP of the last proxy traversed.

Reply:
Yep, see above